packages/core/src/rules/common/hide-internals.ts

```diff
@@ -0,0 +1,65 @@
+import { Oas3Decorator, Oas2Decorator } from '../../visitors';
+import { isPlainObject } from '../../utils';
+
+const DEFAULT_HIDDEN_TAG = 'x-internal';
+const HIDDEN_TYPES = [
+  'PathItem',
+  'Operation',
+  'Schema',
+  'Response',
+  'RequestBody',
+  'Example',
+  'MediaType',
+  'Server',
+  'Callback',
+  'Parameter',
+];
+
+export const HideInternals: Oas3Decorator | Oas2Decorator = ({ hideTag }) => {
```

Comment:
there are a lot of conditions, please cover all of them with tests

Reply:
`tagToHide` option is covered with the test.